### PR TITLE
PXC-2621: Assert on COMMIT/ROLLBACK WORK AND CHAIN

### PR DESCRIPTION
**Pull request by kamil-holubicki**
PXC-2621: Assert on COMMIT/ROLLBACK WORK AND CHAIN

https://jira.percona.com/browse/PXC-2621

Added missing cleanup of wsrep client/transaction state after commit/rollback transaction and before beginning new transaction if COMMIT/ROLLBACK WORK AND CHAIN was executing.

Not having this caused that transaction was in 'commited' state and new transaction expected uninitialized one. In regular COMMIT/ROLLBACK case such cleanup would be done just after executing commit/rollback, but in case of COMMIT/ROLLBACK WORK AND CHAIN cleaning client/transaction state after execution of statement was too late.

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=958) <br/>All committers have signed the CLA.

**Comment by kamil-holubicki** [comment]
https://pxc.cd.percona.com/view/PXC%208.0/job/pxc80-test-mtr/156/
MTR tests failures not related to the change.

File: sql/sql_parse.cc [hunk 1]
```diff
@@ -4872,6 +4872,15 @@ int mysql_execute_command(THD *thd, bool first_level) {
...
       /* Begin transaction with the same isolation level. */
       if (tx_chain) {
+#ifdef WITH_WSREP
+        /* We need to cleanup wsrep state before starting
+           new transaction. If 'regular' commit was issued,
+           it would be done in caller function wsrep_mysql_parse()
+           after returning from here.
+           But now we need to do it in between.
+         */
+        wsrep_after_statement(thd);
```
Comment (kennt-percona) [inline sql/sql_parse.cc:4882]: Do we need to do this also for 'ROLLBACK AND CHAIN'?

**Review by kennt-percona — APPROVED** [review]
lgtm

**Pull request merged**